src/fileoverview_comment_transformer.ts

```diff
@@ -0,0 +1,77 @@
+ */
+export function transformFileoverviewComment(context: ts.TransformationContext) {
+  return (sf: ts.SourceFile) => {
+    let comments: ts.SynthesizedComment[] = [];
+    // Use trailing comments because that's what transformer_util.ts creates (i.e. by convention).
+    if (sf.statements.length) comments = ts.getSyntheticTrailingComments(sf.statements[0]) || [];
+
+    let fileoverviewIdx = -1;
+    let parsed: {tags: jsdoc.Tag[]}|null = null;
+    for (let i = comments.length - 1; i >= 0; i--) {
+      const current = jsdoc.parseContents(comments[i].text);
+      if (current !== null && current.tags.some(t => FILEOVERVIEW_COMMENT_MARKERS.has(t.tagName))) {
+        fileoverviewIdx = i;
+        parsed = current;
+        break;
+      }
+    }
+    // Add a @suppress {checkTypes} tag to each source file's JSDoc comment,
+    // being careful to retain existing comments and their @suppress'ions.
+    // Closure Compiler considers the *last* comment with @fileoverview (or @externs or @nocompile)
+    // that has not been attached to some other tree node to be the file overview comment, and
+    // only applies @suppress tags from it.
+    // AJD considers *any* comment mentioning @fileoverview.
+    if (!parsed) {
+      // No existing comment to merge with, just emit a new one.
+      return addNewFileoverviewComment(sf);
+    }
+
+    // Add @suppress {checkTypes}, or add to the list in an existing @suppress tag.
+    // Closure compiler barfs if there's a duplicated @suppress tag in a file, so the tag must
+    // only appear once and be merged.
+    const {tags} = parsed;
+    const suppressTag = tags.find(t => t.tagName === 'suppress');
+    if (suppressTag) {
+      const suppressions = suppressTag.type || '';
+      const suppressionsList = suppressions.split(',').map(s => s.trim());
+      if (suppressionsList.indexOf('checkTypes') === -1) {
+        suppressionsList.push('checkTypes');
+      }
+      suppressTag.type = suppressionsList.join(',');
+    } else {
+      tags.push({
+        tagName: 'suppress',
+        type: 'checkTypes',
+        text: 'checked by tsc',
+      });
+    }
+    const commentText = jsdoc.toStringWithoutStartEnd(tags);
+    comments[fileoverviewIdx].text = commentText;
+    // sf does not need to be updated, synthesized comments are mutable.
+    return sf;
+  };
+}
+
+function addNewFileoverviewComment(sf: ts.SourceFile): ts.SourceFile {
+  const commentText = jsdoc.toStringWithoutStartEnd([
+    {tagName: 'fileoverview', text: 'added by tsickle'},
+    {tagName: 'suppress', type: 'checkTypes', text: 'checked by tsc'},
+  ]);
+  let syntheticFirstStatement = ts.createNotEmittedStatement(sf);
+  syntheticFirstStatement = ts.addSyntheticTrailingComment(
+      syntheticFirstStatement, ts.SyntaxKind.MultiLineCommentTrivia, commentText, true);
+  return ts.updateSourceFileNode(sf, [syntheticFirstStatement, ...sf.statements]);
```

Comment:
This might break in combination with other transformations especially decorators. Consider using `updateSourceFileNode` from `transformer_util` here if you see errors on TAP.